blockscout-ens/bens-proto/proto/bens.proto

```diff
@@ -40,101 +62,99 @@ message DomainEvent {
-  uint64 chain_id = 2;
-  // Specifies how much information about domain should be returned.
-  DomainView view = 3;
+  int64 chain_id = 2;
 }
 
 message ListDomainEventsRequest {
-  // Unique id for the domain, also known as nodehash
-  string domain_id = 1;
+  // Name of domain, for example vitalik.eth
+  string name = 1;
   // The chain (network) where domain search should be done
-  uint64 chain_id = 2;
+  int64 chain_id = 2;
+  // Sorting field. Default is `timestamp`
+  string sort = 4;
+  // Order direction. Default is ASC
+  OrderDirection order = 5;
 }
 
 message ListDomainEventsResponse {
-  repeated DomainEvent domain_events = 1;
+  repeated DomainEvent items = 1;
+  Pagination pagination = 2;
 }
 
-message ResolveDomainNameRequest {
+message LookupDomainNameRequest {
   // Name of domain, for example vitalik.eth
   string name = 1;
   // The chain (network) where domain search should be done
-  uint64 chain_id = 2;
-  // Specifies how much information about domain should be returned.
-  DomainView view = 3;
+  int64 chain_id = 2;
   // Filtering field to remove expired domains
-  bool only_active = 4;
+  bool only_active = 3;
+  // Sorting field. Default is `registration_date`
+  string sort = 4;
+  // Order direction. Default is ASC
+  OrderDirection order = 5;
 }
 
-message ResolveDomainNameResponse {
-  // Basic domain info.
-  // May be empty in case if domain name haven't been resolved.
-  // The caller should check that `domain.id` is not zero.
-  Domain domain = 1;
-  // List of domain events. Is empty, if BASIC domain view have been requested.
-  repeated DomainEvent domain_events = 2;
+message LookupDomainNameResponse {
+  // List of domains that resolved to or owned by requested address
+  // Sorted by relevance, so first address could be displayed as main resolved address
+  repeated Domain items = 1;
+  Pagination pagination = 2;
 }
 
-
-message ResolveAddressRequest {
+message LookupAddressRequest {
   // Address of EOA or contract
   string address = 1;
   // The chain (network) where domain search should be done
-  uint64 chain_id = 2;
-  // Specifies how much information about corresponding domain should be returned.
-  DomainView view = 3;
+  int64 chain_id = 2;
   // Include domains resolved to the address
   bool resolved_to = 4;
   // Include domains owned by the address
   bool owned_by = 5;
   // Filtering field to remove expired domains
   bool only_active = 6;
+  // Sorting field. Default is `registration_date`
+  string sort = 7;
+  // Order direction. Defaut is ASC
+  OrderDirection order = 8;
 }
 
-message ResolveAddressResponse {
-  // List of domains that resolved to requested address
+message LookupAddressResponse {
+  // List of domains that resolved to or owned by requested address
   // Sorted by relevance, so first address could be displayed as main resolved address
-  repeated Domain resolved = 1;
-  // List of domains owned by requested address
-  repeated Domain owned = 2;
```

Comment:
We we decided to combine it into one field?

Reply:
yes, single `items` field is more appropriate here

Reply:
for instance owned can be just copy of resolved (If I own domain, I can easily set resolved_address to my addr). therefore we need to remove duplicates. frontend can distinguish between resolved and owned by simply filtering `const resolved = items.filter((item) => item.resolved_address == address)`. also single `items` field is appropriate way to do pagination